src/java/org/apache/cassandra/utils/concurrent/LightweightRecycler.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils.concurrent;
+
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+
+interface LightweightRecyclerPoolHolder<T>
+{
+    ArrayDeque<T> get();
+}
+
+/**
+ * A simple thread local object reuse facility with limited capacity and no attempt at rebalancing pooling between
+ * threads. This is meant to be put in place where churn is high, but single object allocation and footprint are not
+ * so high to justify a more sophisticated approach.
+ * <p>
+ * <b>Internal use only</b>
```

Comment:
What does "internal" mean here?

Reply:
I do not know what the intention of that comment was but I've removed it because the majority of our classes are for internal use only.